breadth_first_traversal/BreadthFirstTraversal.java

```diff
@@ -0,0 +1,64 @@
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Set;
+
+public class BreadthFirstSearch {
+
+    // Array  of lists for Adjacency List Representation
+    public static LinkedList<Integer>[] adj;
```

Comment:
You are saying that the adjacency list may contain only integer items, while they can be anything. They can be names, addresses, tweets, a state of N-Queen board, etc. You just need to define that they are somehow linked.

So, instead of using `Integer`, you can go with generic objects.